bot/cogs/clean.py

```diff
@@ -129,6 +130,19 @@ def predicate_regex(message: Message) -> bool:
                 if not self.cleaning:
                     return
 
+                # If we are looking for specific message.
+                if until_message:
+                    # Since we will be using `delete_messages` method of a TextChannel and we need message objects to
+                    # use it as well as to send logs we will start appending messages here instead adding them from
+                    # purge.
+                    messages.append(message)
+                    # we could use ID's here however in case if the message we are looking for gets deleted,
+                    # we won't have a way to figure that out thus checking for datetime should be more reliable
+                    if message.created_at <= until_message.created_at:
+                        # means we have found the message until which we were supposed to be deleting.
+                        message_ids.append(message.id)
+                        break
```

Comment:
If the current message is older than `until_message`, then the current message should not be appended and deleted, as it's too far back. Something like this, but maybe you can come up with something more elegant:

```py
if message.created_at < until_message.created_at:
    break

messages.append(message)

if message.id == until_message.id:
    message_ids.append(message.id)
    break
```